pkg/local_object_storage/shard/gc.go

```diff
@@ -206,108 +214,83 @@ func (s *Shard) removeGarbage() {
 	}
 }
 
-func (s *Shard) collectExpiredObjects(e Event) {
-	epoch := e.(newEpoch).epoch
-	log := s.log.With(zap.Uint64("epoch", epoch))
-
-	log.Debug("started expired objects handling")
-
-	expired, err := s.getExpiredObjects(e.(newEpoch).epoch, func(typ object.Type) bool {
-		return typ != object.TypeLock && typ != object.TypeTombstone
-	})
-	if err != nil {
-		log.Warn("iterator over expired objects failed", zap.Error(err))
+func (s *Shard) collectExpiredObjects() {
+	epoch := s.gc.currentEpoch.Load()
+	doneUpTo := s.gc.processedEpoch.Load()
+	if s.info.Mode.NoMetabase() || doneUpTo == epoch {
 		return
 	}
-	if len(expired) == 0 {
-		log.Debug("no expired objects")
+	if doneUpTo > epoch {
+		s.log.Warn("current epoch is less than the last processed epoch",
```

Comment:
It'd be strange to see this, but at least GC should be mentioned then, otherwise this message doesn't have a lot of meaning.